django/db/models/expressions.py

```diff
@@ -369,7 +369,11 @@ def __eq__(self, other):
 
     def __hash__(self):
         path, args, kwargs = self.deconstruct()
-        return hash((path,) + args + tuple(kwargs.items()))
+        kwargs = kwargs.copy()
+        output_field = type(kwargs.pop('output_field', None))
+        return hash((path, output_field) + args + tuple([
+            (key, tuple(value)) if isinstance(value, list) else (key, value) for key, value in kwargs.items()
```

Comment:
I'd put `for key, value in kwargs.items()` on the next line to decrease length a bit.